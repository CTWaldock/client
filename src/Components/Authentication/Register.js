import React, { Fragment } from 'react'
import Form from '../../UI/Form'
import FormField from '../../UI/FormField'
import Button from '../../UI/Button'
import { StyledLink, ErrorMessage, Center } from './Auth.styles.js'



const Register = (props) => {
   const { registerError, register, url} = props

       
       return (
           <Fragment>
               <Center>
                    <Form onSubmit={(e) => register(e, url)} title="Register an account!">
                        <FormField label="Email" name="email" type="email" onChange="" />
                        <FormField label="Password" name="password" type="password" onChange="" />
                        <ErrorMessage>{registerError}</ErrorMessage>

                        <Button className="formButton" text="Create account" />
                        <StyledLink to="/login">Already have an account?</StyledLink>
                    </Form>
                </Center>
            </Fragment>
   )
}

export default Register