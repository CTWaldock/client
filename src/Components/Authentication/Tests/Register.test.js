import React from 'react'
import Register from '../Register'
import renderer from 'react-test-renderer'
import {BrowserRouter as Router} from 'react-router-dom'
// import localStorage from '../../localStorage'

// window.localStorage = localStorage

describe('Register', () => {
    it('Should render the way we expect', () => {
        
        
        const tree = renderer.create(
            <Router>
                <Register registerError='something' register='something' url='something' />
            </Router>
        )
        expect(tree.toJSON()).toMatchSnapshot()
    })
})
