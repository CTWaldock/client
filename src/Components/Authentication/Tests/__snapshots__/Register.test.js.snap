// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Should render the way we expect 1`] = `
<div
  className="sc-VigVT gUteem"
>
  <form
    className="sc-gzVnrw eqCBFZ"
    id={undefined}
    onSubmit={[Function]}
  >
    <h3
      className="sc-htoDjs fbZTTv"
    >
      Register an account!
    </h3>
    <label
      className="sc-dnqmqq gxITcO"
    >
      First Name
      <span />
    </label>
    <input
      autoComplete="nope"
      autoFocus={null}
      className="sc-iwsKbI kxDmNQ"
      name="firstName"
      onChange={null}
      placeholder={undefined}
      step={null}
      type="text"
    />
    <label
      className="sc-dnqmqq gxITcO"
    >
      Last Name
      <span />
    </label>
    <input
      autoComplete="nope"
      autoFocus={null}
      className="sc-iwsKbI kxDmNQ"
      name="lastName"
      onChange={null}
      placeholder={undefined}
      step={null}
      type="text"
    />
    <label
      className="sc-dnqmqq gxITcO"
    >
      Email
      <span />
    </label>
    <input
      autoComplete="nope"
      autoFocus={null}
      className="sc-iwsKbI kxDmNQ"
      name="email"
      onChange={null}
      placeholder={undefined}
      step={null}
      type="email"
    />
    <label
      className="sc-dnqmqq gxITcO"
    >
      Password
      <span />
    </label>
    <input
      autoComplete="nope"
      autoFocus={null}
      className="sc-iwsKbI kxDmNQ"
      name="password"
      onChange={null}
      placeholder={undefined}
      step={null}
      type="password"
    />
    <p
      className="sc-gqjmRU cKmdHu"
    >
      something
    </p>
    <button
      className="sc-jTzLTM ffePeW"
    >
      Create account
    </button>
    <a
      className="sc-gZMcBi htfulI"
      href="/login"
      onClick={[Function]}
    >
      Already have an account?
    </a>
    <div />
  </form>
</div>
`;
